tanner/emulators/rfi.py

```diff
@@ -56,20 +52,20 @@ async def download_file(self, path):
                     rfile.write(data.encode('utf-8'))
         return file_name
 
-    def download_file_ftp(self, url):
+    async def download_file_ftp(self, url):
         host = url.host
-        ftp_path = url.path.rsplit('/', 1)[0][1:]
+        ftp_path = url.path
         name = url.name
         try:
-            ftp = ftplib.FTP(host)
-            ftp.login()
-            ftp.cwd(ftp_path)
+            ftp = aioftp.Client()
+            await ftp.connect(host)
+            await ftp.login()
             tmp_filename = name + str(time.time())
             file_name = hashlib.md5(tmp_filename.encode('utf-8')).hexdigest()
-            with open(os.path.join(self.script_dir, file_name), 'wb') as ftp_script:
-                self.logger.debug('Saving the FTP file as %s', os.path.join(self.script_dir, file_name))
-                ftp.retrbinary('RETR %s' % name, ftp_script.write)
-        except ftplib.all_errors as ftp_errors:
+            self.logger.debug('Saving the FTP file as %s', os.path.join(self.script_dir, file_name))
+            await ftp.download(ftp_path, os.path.join(self.script_dir, file_name), write_into=True)
+            await ftp.quit()
+        except (AIOFTPException, StatusCodeError, PathIsNotAbsolute, PathIOError, NoAvailablePort) as ftp_errors:
```

Comment:
AIOFTPException is the base class. All other exceptions are inherited from that class. Why you need to specify all of them? Which functions raises them specifically?

Reply:
going with AIOFTPExeption was my initial intention but during testing (after committing) was raising StatusCodeError.
This error was raised from test_ftp_download().